src/main/java/com/dev/movietheatre/security/impl/CustomUserDetailsService.java

```diff
@@ -0,0 +1,33 @@
+package com.dev.movietheatre.security.impl;
+
+import com.dev.movietheatre.model.User;
+import com.dev.movietheatre.service.UserService;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String userName) throws UsernameNotFoundException {
+        User user = userService.findByEmail(userName)
+                .orElseThrow(() -> new UsernameNotFoundException("User was not found!"));
```

Comment:
add method input data to message to make more informative

Reply:
done